### PR TITLE
Migrate list of table prefixes to Predicate in CachingDirectoryListener

### PR DESCRIPTION
**Pull request by marcinsbd**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java [hunk 1]
```diff
@@ -65,6 +65,10 @@ public CachingDirectoryLister(HiveConfig hiveClientConfig)
 
     public CachingDirectoryLister(Duration expireAfterWrite, DataSize maxSize, List<String> tables, Predicate<FileEntry> filterPredicate)
     {
+        requireNonNull(expireAfterWrite);
+        requireNonNull(maxSize);
+        requireNonNull(tables);
+        requireNonNull(filterPredicate);
```
Comment (Praveen2112) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java:77]: Can we have a meaningfull message for rnn ?

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java (hunk 1 above)
Comment (marcinsbd) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java]: fixed

**Review by ssheikin — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java [hunk 2]
```diff
@@ -83,12 +82,19 @@ public CachingDirectoryLister(
...
                 .recordStats()
                 .build();
-        this.tablePrefixes = tables.stream()
-                .map(CachingDirectoryLister::parseTableName)
-                .collect(toImmutableList());
+        this.tablePredicate = isIncluded(tables);
         this.filterPredicate = filterPredicate;
     }
 
+    private static Predicate<SchemaTableName> isIncluded(List<String> tables)
```
Comment (ssheikin) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java:89]: rename to `matches`

**Comment by kokosing** [comment]
Please merge https://github.com/trinodb/trino/pull/22704 and make sure that common commits are exactly the same. Please ping me when it is done and I will merge it.

**Review by ssheikin — APPROVED** [review]

**Pull request merged**